internal/comparetest/golden/golden_test.go

```diff
@@ -33,7 +33,14 @@ func TestWriteMetrics(t *testing.T) {
 	metricslice.CopyTo(metrics.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics())
 
 	actualFile := filepath.Join(t.TempDir(), "metrics.json")
-	require.NoError(t, WriteMetrics(actualFile, metrics))
+
+	// Overwrite the callback, to aviod an actual failure.
+	var failed bool
+	writeMetricsCallback = func(t *testing.T) {
+		failed = true
+	}
+	require.NoError(t, WriteMetrics(t, actualFile, metrics))
```

Comment:
Maybe we can use `require.Panics` and don't mutate the callback?

Reply:
I can't figure out a way to make this work without making `WriteMetrics` panic, which seems like it would be much more confusing to devs who are using it as intended.